pkg/storage/unified/search/bleve_test.go

```diff
@@ -766,517 +768,323 @@ func setupBleveBackend(t *testing.T, fileThreshold int, cacheTTL time.Duration,
-		# TYPE index_server_open_indexes gauge
-		index_server_open_indexes{index_storage="memory"} 0
-		index_server_open_indexes{index_storage="file"} 1
-	`), "index_server_open_indexes"))
-
-	backend3.CloseAllIndexes()
-
-	// again now RV > 100. Should still rebuild the index due to the rebuild flag
-	backend4, reg4 := setupBleveBackend(t, 5, time.Nanosecond, tmpDir)
-	idx, err = backend4.BuildIndex(context.Background(), ns, 10 /* file based */, 101, nil, "test", indexTestDocs(ns, 1002, 100), nil, true, false)
-	require.NoError(t, err)
-
-	// Verify that we've re-built the index. There should be 1002 documents
-	cnt, err = idx.DocCount(context.Background(), "")
-	require.NoError(t, err)
-	require.Equal(t, int64(1002), cnt)
-
-	require.NoError(t, testutil.GatherAndCompare(reg4, bytes.NewBufferString(`
-		# HELP index_server_open_indexes Number of open indexes per storage type. An open index corresponds to single resource group.
-		# TYPE index_server_open_indexes gauge
-		index_server_open_indexes{index_storage="memory"} 0
-		index_server_open_indexes{index_storage="file"} 1
-	`), "index_server_open_indexes"))
-
-	backend4.CloseAllIndexes()
-}
-
-func TestFileIndexIsIgnoredIfRebuildFlagIsTrueWithSearchAfterWrite(t *testing.T) {
-	ns := resource.NamespacedResource{
-		Namespace: "test",
-		Group:     "group",
-		Resource:  "resource",
-	}
-
-	tmpDir := t.TempDir()
+func TestBuildIndex(t *testing.T) {
+	t.Run("closeAllIndexes should close all open indexes", func(t *testing.T) {
+		ns := resource.NamespacedResource{
+			Namespace: "test",
+			Group:     "group",
+			Resource:  "resource",
+		}
+		ns2 := resource.NamespacedResource{
+			Namespace: "test2",
+			Group:     "group",
+			Resource:  "resource",
+		}
 
-	backend1, reg1 := setupBleveBackend(t, 5, time.Nanosecond, tmpDir)
-	_, err := backend1.BuildIndex(context.Background(), ns, 10 /* file based */, 100, nil, "test", indexTestDocs(ns, 10, 100), nil, true, true)
-	require.NoError(t, err)
+		tmpDir := t.TempDir()
+		backend1, reg1 := setupBleveBackend(t, 5, time.Nanosecond, tmpDir)
+		_, err := backend1.BuildIndex(context.Background(), ns, 10 /* file based */, 100, nil, "test", indexTestDocs(ns, 10, 100), nil, false, false)
+		require.NoError(t, err)
+		_, err = backend1.BuildIndex(context.Background(), ns2, 1 /* memory based */, 100, nil, "test", indexTestDocs(ns, 10, 100), nil, false, false)
+		require.NoError(t, err)
 
-	// Verify one open index.
-	require.NoError(t, testutil.GatherAndCompare(reg1, bytes.NewBufferString(`
+		// Verify two open indexes.
+		require.NoError(t, testutil.GatherAndCompare(reg1, bytes.NewBufferString(`
 		# HELP index_server_open_indexes Number of open indexes per storage type. An open index corresponds to single resource group.
 		# TYPE index_server_open_indexes gauge
-		index_server_open_indexes{index_storage="memory"} 0
+		index_server_open_indexes{index_storage="memory"} 1
 		index_server_open_indexes{index_storage="file"} 1
 	`), "index_server_open_indexes"))
```

Comment:
Can we extract this check into simpler `verifyOpenIndexes(t, registry, inMemory, fileBased)` method and call with expected inmemory/file values?